ts/client/database/databaseEngine.ts

```diff
@@ -0,0 +1,129 @@
+import { ZERO_BYTES32 } from "../../constants";
+import { TreeAtLevelIsFull } from "../../exceptions";
+import { Hashable, Vacant } from "../../interfaces";
+import { Hasher } from "../../tree";
+import { DBTree } from "../../tree/dbTree";
+import { Leaf, LeafFactoryFunc } from "../../tree/leaves/Leaf";
+import { StorageEngine, WithWitness } from "../storageEngine/interfaces";
+
+export interface Entry<Item> {
+    itemID: number;
+    item: Item;
+}
```

Comment:
This should be moved to `storageEngine/interfaces` and used here and in `MemoryStorageEngine`